### PR TITLE
feature: update documentation format

### PR DESCRIPTION
**Pull request by ju-Skinner**
# Description

* @pine-ds/doc-components:  Introduce new component library for Documentation
* Refactor existing documentation mdx files to use new file format and components
* 

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

- [X] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests you've added and run to verify your changes.
Provide instructions so that we can reproduce.
Please also list any relevant details for your test configuration.

- [ ] unit tests
- [ ] e2e tests
- [ ] accessibility tests
- [ ] tested manually
- [ ] other:

**Test Configuration**:

- Pine versions:
- OS:
- Browsers:
- Screen readers:
- Misc:

# Checklist:

If not applicable, leave options unchecked.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

### CODE REVIEW

File: libs/core/src/components/pds-image/docs/pds-image.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+</DocCanvas>
+
+### Sizes and Srcset
+
+An image with `srcset` and `sizes` props set. Refresh the browser at `400px` and below and the browser will select the smaller image to render.
+
+<DocCanvas client:only
+  mdxSource={{
+    react: `
+<pds-image
```
Comment (pixelflips) [inline libs/core/src/components/pds-image/docs/pds-image.mdx]: I think this should be `PdsImage`

File: libs/core/src/components/pds-input/docs/pds-input.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+}}>
+  <pds-input component-id="pds-input-help-example" label="Name" type="email" helper-message="Please use the correct format" value="Frank Dux"></pds-input>
+</DocCanvas>
+
+### Invalid
+<p>When `error-message` is present, the input displays the error message and provides a visual treatment.</p>
+
+<DocCanvas client:only
+  mdxSource={{
+    react: '<PdsInput componentId="pds-input-error-example" label="Name" type="email" error-message="Please provide a valid email address" value="Frank Dux"></PdsInput>',
```
Comment (pixelflips) [inline libs/core/src/components/pds-input/docs/pds-input.mdx]: I think this should be `errorMessage`.

File: libs/core/src/components/pds-progress/docs/pds-progress.mdx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+}}>
+  <pds-progress component-id="percent" label="percent" percent="25"></pds-progress>
+</DocCanvas>
+
+### Show Percent
+
+Displays the `percent` value as text beside the progress.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsProgress componentId="show-percent" label="Show percent" percent="50" show-percent={true}></PdsProgress>`,
```
Comment (pixelflips) [inline libs/core/src/components/pds-progress/docs/pds-progress.mdx]: I think this should be `showPercent`

File: libs/core/src/components/pds-progress/docs/pds-progress.mdx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+}}>
+  <pds-progress component-id="animated" label="Animated " percent="75" animated></pds-progress>
+</DocCanvas>
+
+### Fill Color
+
+Allows for `pds-progress` to be filled with a custom color. Accepts any valid color token or CSS color value.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsProgress componentId="fillcolor" label="Fill color" percent="95" fill-color="#86D5BC"></PdsProgress>`,
```
Comment (pixelflips) [inline libs/core/src/components/pds-progress/docs/pds-progress.mdx]: I think this should be `fillColor`

File: libs/core/src/components/pds-progress/docs/pds-progress.mdx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  <pds-progress component-id="fillcolor" label="Fill color" percent="95" fill-color="#86D5BC"></pds-progress>
+</DocCanvas>
+
+### Tooltip
+
+A tooltip can be added by combining the `progress` component with a [tooltip](/docs/components-tooltip--default).
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsTooltip content="50%" has-arrow="false" placement="top" style={{ width: '100%' }}>
+    <PdsProgress component-id="tooltip-demo" label="Progress w/ tooltip" percent="50"></PdsProgress>
```
Comment (pixelflips) [inline libs/core/src/components/pds-progress/docs/pds-progress.mdx]: `componentId` here.

File: libs/core/src/components/pds-progress/docs/pds-progress.mdx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  </PdsTooltip>`,
+  webComponent: `<pds-tooltip content="50%" has-arrow="false" placement="top" style={{ width: '100%' }}>
+    <pds-progress component-id="tooltip-demo" label="Progress w/ tooltip" percent="50"></pds-progress>
```
Comment (pixelflips) [inline libs/core/src/components/pds-progress/docs/pds-progress.mdx:72]: `componentId` here.

File: libs/core/src/components/pds-radio/docs/pds-radio.mdx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+}}>
+  <pds-radio component-id="disabled" label="Label" disabled />
+</DocCanvas>
+
+### Message
+
+Message is a string of text that displays below the label text.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsRadio componentId="message1" label="Label" helper-message="This is short message text." />`,
```
Comment (pixelflips) [inline libs/core/src/components/pds-radio/docs/pds-radio.mdx]: `helperMessage` here.

File: libs/core/src/components/pds-radio/docs/pds-radio.mdx [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+</DocCanvas>
+
+### Invalid
+
+Invalid displays the radio in invalid state such as when required but is not checked.
+
+An optional error message can be displayed for additional instructions, displayed independently from the helper message.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsRadio componentId="invalid" label="Label" invalid error-message="This is a short error message" />`,
```
Comment (pixelflips) [inline libs/core/src/components/pds-radio/docs/pds-radio.mdx]: `errorMessage` here.

File: libs/core/src/components/pds-sortable/docs/pds-sortable.mdx [hunk 9]
```diff
@@ -0,0 +1,301 @@
+import { DocArgsTable, DocCanvas } from '@pine-ds/doc-components'
+
+# Sortable
+
+Sortable is for creating containers of items that can be sorted via drag and drop functionality.
+
+## Properties
+
+<DocArgsTable componentName='pds-sortable' />
+
```
Comment (pixelflips) [inline libs/core/src/components/pds-sortable/docs/pds-sortable.mdx:11]: nitpick, extra space.

File: libs/core/src/components/pds-sortable/docs/pds-sortable.mdx [hunk 10]
```diff
@@ -0,0 +1,301 @@
...
+    <pds-sortable-item>Item 3</pds-sortable-item>
+  </pds-sortable>
+</DocCanvas>
+
+### Bordered
+
+When `border` is set to `true`, the component will render a border around all items.
+
+<DocCanvas client:only mdxSource={{
+  react:`<PdsSortable border component-id="bordered">
```
Comment (pixelflips) [inline libs/core/src/components/pds-sortable/docs/pds-sortable.mdx]: `componentId` here

File: libs/core/src/components/pds-sortable/docs/pds-sortable.mdx [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+</DocCanvas>
+
+### Actions
+
+The `enable-actions` property in the `pds-sortable-item` component determines whether the `sortable-item-actions` slot should be enabled.
+
+The `sortable-item-actions` slot is provided to place content within the `pds-sortable-item__actions` element as children, but this slot is only rendered if the `enable-actions` property is set to `true`.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsSortable component-id="actions">
```
Comment (pixelflips) [inline libs/core/src/components/pds-sortable/docs/pds-sortable.mdx]: `componentId` here

File: libs/core/src/components/pds-sortable/docs/pds-sortable.mdx [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+  <PdsSortableItem enable-actions>
```
Comment (pixelflips) [inline libs/core/src/components/pds-sortable/docs/pds-sortable.mdx]: `enableActions` here and the following items below.

File: libs/core/src/components/pds-sortable/docs/pds-sortable.mdx [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+  </pds-sortable>
+</DocCanvas>
+
+### Demo
+
+The following is a demo of how all the properties and slots can be used together to create a sortable list.
+
+<DocCanvas client:only mdxSource={{
+  react: `<PdsSortable border={true} dividers={true} componentId="demo">
+  <PdsSortableItem enable-actions handle>
```
Comment (pixelflips) [inline libs/core/src/components/pds-sortable/docs/pds-sortable.mdx:213]: `enableActions` here and the following items below.

File: libs/core/src/components/pds-progress/stories/pds-progress.docs.mdx [hunk 14]
```diff
@@ -3,73 +3,8 @@ import { html, render } from 'lit';
```
Comment (pixelflips) [inline libs/core/src/components/pds-progress/stories/pds-progress.docs.mdx:2]: I think progress is missing the playground

File: libs/core/src/components/pds-radio/stories/pds-radio.docs.mdx [hunk 15]
```diff
@@ -3,86 +3,8 @@ import { html, render } from 'lit';
```
Comment (pixelflips) [inline libs/core/src/components/pds-radio/stories/pds-radio.docs.mdx:2]: i think radio is missing the playground

File: libs/core/src/components/pds-tooltip/docs/pds-tooltip.mdx [hunk 16]
```diff
@@ -0,0 +1,116 @@
...
+
+<DocCanvas client:only
+  mdxSource={{
+    react: `
+      <PdsTooltip hasArrow="true" placement="right">
+        text
+        <div slot="content">
+          <p>this is a sentence in a tooltip.</p>
+          <p>this is a sentence in a tooltip.</p>
+        </div>
```
Comment (pixelflips) [inline libs/core/src/components/pds-tooltip/docs/pds-tooltip.mdx:78]: Can prob be handled in a follow up but looks like the HTML content is inheriting some colors from somewhere, which is odd...

Screenshot 2023-12-06 at 4 42 48 PM

**Review by pixelflips — CHANGES_REQUESTED** [review]
Overall I dig this approach. A few comments that are mostly related to typos and some missing playground sections.

If the playground sections are intentional just mark those comments as resolved and ignore.

**Comment by ju-Skinner** [comment]
Forgive me, I guess I created the PR prematurely, but if we see something feel free to fix it.

File: libs/core/src/components/pds-tooltip/docs/pds-tooltip.mdx (hunk 16 above)
Comment (ju-Skinner) [reply libs/core/src/components/pds-tooltip/docs/pds-tooltip.mdx:78]: Here is the style that is affecting it. I'm just not sure as to why. Looking into this now
image

Comment (ju-Skinner) [reply libs/core/src/components/pds-tooltip/docs/pds-tooltip.mdx:78]: You can see removing color, resolves the issue
image

Comment (pixelflips) [reply libs/core/src/components/pds-tooltip/docs/pds-tooltip.mdx:78]: seems odd that what I am assuming are the storybook styles would affect/override the component styles.

**Comment by ju-Skinner** [comment]
@Kajabi/dss-devs I have consolidated and restructured the commits. Please re-review. Thanks

**Review by pixelflips — APPROVED** [review]
Outside of the non-blocking previously mentioned tooltip color issue, this is awesome! LGTM!

**Review by anechol — APPROVED** [review]
Love it

**Review by QuintonJason — APPROVED** [review]
The docs are working and the stencil components are being pulled in as well. Great work 🏆

**Pull request merged**